pkg/workloadmeta/store.go

```diff
@@ -275,6 +275,48 @@ func (s *store) GetKubernetesPod(id string) (*KubernetesPod, error) {
 	return entity.(*KubernetesPod), nil
 }
 
+// GetProcess implements Store#GetProcess.
+func (s *store) GetProcess(pid int32) (*Process, error) {
+	id := string(pid)
+
+	entity, err := s.getEntityByKind(KindProcess, id)
+	if err != nil {
+		return nil, err
+	}
+
+	return entity.(*Process), nil
+}
+
+// ListProcesses implements Store#ListProcesses.
+func (s *store) ListProcesses() []*Process {
+	entities := s.listEntitiesByKind(KindProcess)
+
+	// Not very efficient
+	processes := make([]*Process, 0, len(entities))
+	for i := range entities {
+		processes = append(processes, entities[i].(*Process))
+	}
+
+	return processes
+}
+
+// ListProcessesWithFilter implements Store#ListProcessesWithFilter
+func (s *store) ListProcessesWithFilter(filter ProcessFilterFunc) []*Process {
+	entities := s.listEntitiesByKind(KindProcess)
+
+	// Not very efficient
```

Comment:
Sorry, this threw me for a loop a bit. Can you elaborate on why this isn't efficient and add a todo?

Alternatively we can remove this comment, unless I'm missing something I don't think what you have isn't efficient

Reply:
It was there for containers but I can try to explain.

Here we first listEntitiesByKind which returns a slice with all the processes in it while we could avoid this allocation.